### DISCUSSION TITLE
WordPress: bypass cache on querystring, does disabling this impacts somewhere?

### DISCUSSION THREAD

**Discussion by SGr33n** [EasyEngine/easyengine#1747] [Support Requests]
Hi,
I can see that one of the suggested parts of the nginx configuration file for WordPress running fastcgi cache have this directive:
```
	if ($query_string != "") {
		set $skip_cache 1;
	}   
```
I'm thinking to disable this because all links from Facebook automatically have a querystring, also the AMP pages have a querystring and all that pages would bypass the cache.

What do you think about this?

Thanks!

**Comment by mrrobot47** [comment]
It is not ideal to completely ignore query string and have everything cached. At times, there could be things like search which could give stale results when cached due to scenarios where the query string is ignored. But that could be absolutely fine on some sites whereas not so much on other sites. This highly depends and varies from site to site.

If there are only a few particular query strings that you wish to not bypass cache, like AMP pages, here's what can be done:

```nginx
	if ($query_string != "") {
		set $skip 1;
	}

	# Don't bypass if the query string contains "?amp=1"
	if ( $arg_amp ) {
		set $skip 0;
	}
```
Similarly, you could have rules for other things where you wish to not bypass the cache.

Once, you have done the required edits to the nginx config do not forget to reload nginx:
```bash
ee site reload <site-name> --nginx
```

**Comment by SGr33n** [comment]
Thanks for your suggestion. This works also where the amp querystring is just ?amp, right (without any value)?
